crates/uv/src/commands/tool/install.rs

```diff
@@ -599,17 +605,40 @@ pub(crate) async fn install(
         }
     };
 
+    let entrypoints = with_executables_from
+        .iter()
+        .map(|source| match source {
+            RequirementsSource::Package(name) => {
+                if let RequirementsTxtRequirement::Named(requirement) = name {
+                    Ok(requirement.name.clone())
+                } else {
+                    bail!(
+                        "Expected a named requirement, but got: {}",
+                        source.to_string().cyan()
+                    )
+                }
+            }
+            _ => bail!(
+                "Expected a package requirement, but got: {}",
+                source.to_string().cyan()
+            ),
+        })
+        .collect::<Result<Vec<_>>>()?;
```

Comment:
I wonder if we should resolve these to `PackageName` over in `uv/src/lib.rs` instead so you don't need to do this here? Over there, we know they're package names.